File: SuperiorPackGroup/BLL/ItemPoolBLL.vb
Imports System.Text
Imports DXDAL.SPGData
Imports DevExpress.Xpo

Public Class ItemPoolBLL

    Public Shared Function ValidateRecord(ByVal poolCode As String, ByVal customerID As Integer?) As Boolean

        If String.IsNullOrEmpty(poolCode) Then
            Return False
        End If
        If Not customerID.HasValue Then
            Return False
        End If

        Return True

    End Function

    Public Shared Function DeleteItemPool(ByVal poolID As Integer) As Boolean

        Dim pool As ItemPool = Session.DefaultSession.GetObjectByKey(Of ItemPool)(poolID)

        If PoolBomBLL.PoolUsedAsBom(poolID) Then
            MessageBox.Show("You can't delete this item pool there are BOM records associated to it")
        Else
            pool.Delete()
        End If

        Return pool.IsDeleted

    End Function

End Class

File: SuperiorPackGroup/BLL/ItemPoolBLL.cs
//INSTANT C# NOTE: Formerly VB project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;

using System.Text;
using DXDAL.SPGData;
using DevExpress.Xpo;

namespace SuperiorPackGroup
{
	public class ItemPoolBLL
	{

		public static bool ValidateRecord(string poolCode, int? customerID)
		{

			if (string.IsNullOrEmpty(poolCode))
			{
				return false;
			}
			if (!customerID.HasValue)
			{
				return false;
			}

			return true;

		}

		public static bool DeleteItemPool(int poolID)
		{

			ItemPool pool = Session.DefaultSession.GetObjectByKey<ItemPool>(poolID);

			if (PoolBomBLL.PoolUsedAsBom(poolID))
			{
				MessageBox.Show("You can't delete this item pool there are BOM records associated to it");
			}
			else
			{
				pool.Delete();
			}

			return pool.IsDeleted;

		}

	}

}